src/keys.ts

```diff
@@ -54,15 +55,27 @@ function parseExtractedKey(key: ExtractedKey, config: Config): TranslationKey {
     const nsSeparatorPos = cleanKey.indexOf(config.nsSeparator);
 
     if (nsSeparatorPos !== -1) {
+      if (key.keyPrefix) {
+        throw new Error(
+          `Do not use the keyPrefix option if you want to use keys with prefixed namespace notation.
+          key: ${cleanKey}
+          keyPrefix: ${key.keyPrefix}`,
+        )
+      }
+
       ns = cleanKey.slice(0, nsSeparatorPos);
       cleanKey = cleanKey.slice(nsSeparatorPos + 1);
     }
   }
 
   let keyPath = Array<string>();
   if (config.keySeparator) {
+    if (key.keyPrefix) {
```

Comment:
Is a key separator always needed? Doesn't i18next have a fallback or something?

Reply:
It is always needed, without it we can't connect prefix with key

https://github.com/i18next/i18next/blob/a6c6680b151fd66e4b4cc2f443aaeaa5d057a453/src/i18next.js#L393-L399


i18next have fallback to `.` but i believe your lib does that too in `parseConfig`
https://github.com/gilbsgilbs/babel-plugin-i18next-extract/blob/5eb8f8bbfaa4f651a662c864f607db0aa455e69e/src/config.ts#L67